azure-communication-ui/chat/src/test/java/com/azure/android/communication/ui/chat/redux/middleware/sdk/ChatActionHandlerUnitTest.kt

```diff
@@ -112,6 +112,81 @@ internal class ChatActionHandlerUnitTest : ACSBaseTestCoroutine() {
             )
         }
 
+    @ExperimentalCoroutinesApi
+    @Test
+    fun chatMiddlewareActionHandler_deleteMessage_then_dispatch_ChatActionDeleteMessage() =
+        runScopedTest {
+            // arrange
+            val messageInfoModel = MessageInfoModel(
+                id = "Message",
+                internalId = "54321",
+                messageType = ChatMessageType.TEXT,
+                content = "hello, world!"
+            )
+
+            val deleteChatMessageCompletableFuture = CompletableFuture<Void>()
+
+            val mockChatService: ChatService = mock {
+                on { deleteMessage(messageInfoModel.id.toString()) } doReturn deleteChatMessageCompletableFuture
+            }
+
+            val chatHandler = ChatActionHandler(mockChatService)
+
+            val action = ChatAction.DeleteMessage(messageInfoModel)
+
+            val mockAppStore = mock<AppStore<ReduxState>> {
+                on { dispatch(any()) } doAnswer { }
+            }
+
+            // act
+            chatHandler.onAction(action, mockAppStore::dispatch)
+
+            deleteChatMessageCompletableFuture.complete(any())
+
+            // assert
+            verify(mockAppStore, times(1)).dispatch(
```

Comment:
I don't think we need to verify Dispatch. (should be covered with redux tests)

However, we should have action->effect tests. So this one should be verifying ChatService.deleteMessage was called correctly.

There should be another test for the Repository Middleware that is
Action->RepositoryMiddleware.onDelete, similar to this one.